spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BinderFactoryAutoConfiguration.java

```diff
@@ -260,14 +260,35 @@ public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
 						Object definedFunction = functionCatalog.lookup(name);
 						Class<?> inputType = inspector.getInputType(definedFunction);
 						Class<?> outputType = inspector.getOutputType(definedFunction);
-						if (Void.class.isAssignableFrom(outputType)) {
-							bind(Sink.class, registry);
+
+						boolean bindDownstream = false;
+						try {
+							Map<String, BindableProvider> bindableProviders = beanFactory.getBeansOfType(BindableProvider.class);
+							Class<?> inputTypeWrapper = inspector.getInputWrapper(definedFunction);
+							if (bindableProviders != null) {
+								Collection<BindableProvider> values = bindableProviders.values();
+								for (BindableProvider bindableProvider : values) {
+									if (bindableProvider.canBind(inputTypeWrapper)) {
+										bindDownstream = true;
```

Comment:
As described in the previous comment, you can actually have annotation (e.g., `BindingProvider`) and annotate the configuration class provided by the binder itself. Then we can simply do:
```
Class<?>[] configurationClasses = bfac.getAll().values().iterator().next().getConfigurationClasses();
		boolean bindingProvider = Stream.of(configurationClasses)
				.filter(clazz -> AnnotationUtils.findAnnotation(clazz, BindingProvider.class) != null)
				.findFirst().isPresent();
. . .
if (!bindingProvider) {
. . .
}
```